src/InitControl.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef INITCONTROL_HPP_INCLUDE
+#define INITCONTROL_HPP_INCLUDE
+
+#include "config.h"
+
+#include <string>
+#include <vector>
+#include <memory>
+
+namespace geopm
+{
+    class InitControl
+    {
+        public:
+            struct m_request_s {
+                std::string name;
+                int domain;
+                int domain_idx;
+                double setting;
+            };
```

Comment:
Does this struct really need to be public? It's not used to interact with the interface and it's only used internally by the implementation (`InitControlImp`).